rir/src/compiler/opt/inline.cpp

```diff
@@ -33,16 +35,24 @@ class TheInliner {
                 Value* staticEnv = nullptr;
 
                 if (auto call = Call::Cast(*it)) {
-                    auto cls = MkFunCls::Cast(call->cls());
-                    if (!cls)
+                    auto mkcls = MkFunCls::Cast(call->cls()->baseValue());
+                    if (!mkcls)
                         continue;
-                    inlinee = cls->fun;
+                    inlinee = mkcls->fun;
                     if (inlinee->argNames.size() != call->nCallArgs())
                         continue;
-                    assert(cls->fun->closureEnv() == Env::notClosed());
-                    staticEnv = cls->env();
+                    staticEnv = mkcls->env();
                 } else if (auto call = StaticCall::Cast(*it)) {
                     inlinee = call->cls();
+                    // if we don't know the closure of the inlinee, we can't
+                    // inline.
+                    if (inlinee->closureEnv() == Env::notClosed() &&
+                        inlinee != function)
+                        continue;
+                    // TODO: honestly I have no clue how namespaces work. For
+                    // now if we hit any env with attribs, we do not inline.
+                    if (ATTRIB(inlinee->closureEnv()->rho))
```

Comment:
Don't they put `R_NilValue` to the slot if there are no attributes? This tests for nullptr